### DISCUSSION TITLE
Unable to Push Branch to Remote Repository - Permission Denied Error

### DISCUSSION THREAD

**Discussion by jacobcbeaudin** [Snowflake-Labs/terraform-provider-snowflake#2709] [General] [answered]
Hello everyone,

I'm encountering an issue while trying to push a new branch to a remote repository and am hoping for some guidance or solutions from the community. Here are the details:

### Issue:
When I attempt to push my local branch using:
```bash
git push --set-upstream origin jacobcbeaudin/update-from-share-docs
```
I receive the following error
```
ERROR: Permission to Snowflake-Labs/terraform-provider-snowflake.git denied to jacobcbeaudin.
fatal: Could not read from remote repository.
```

### Request:
Could anyone help verify if there are additional settings or configurations I might be missing? Any advice on what permissions are required or how to resolve this access denial would be greatly appreciated.

**Comment by sfc-gh-asawicki** [comment, marked as answer]
Hey @jacobcbeaudin. Thanks for reaching out to us.

We do not grant write permissions to the outside contributors inside the repository. The flow to submit proposed changes is to do a fork of the project, submit the changes in your fork, and then open a PR against the main repo (like described here: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/creating-a-pull-request-from-a-fork).
